eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,144 @@
+param(
+  [string] $InputJsonPath = $GenerateInput,
+  [string] $OutputJsonPath = $GenerateOutput,
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+$inputFilePaths = $inputFileContent.changedFiles;
+$inputFilePaths += $inputFileContent.relatedReadmeMdFiles;
+$inputFilePaths = $inputFilePaths | Select-Object -Unique
+
+$changedFilePaths = $inputFilePaths -join "`n";
+Write-Host "List of changed swagger files and related readmes:`n$changedFilePaths`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($inputFilePath in $inputFilePaths) {
+  foreach ($path in $autorestFilesPath) {
+    $fileContent = Get-Content $path
+    $isUpdatedLines = $false
+    $updatedLines = foreach ($line in $fileContent) {
+      $regexForMatchingShaAndPath = "[\/][0-9a-f]{4,40}[\/]$inputFilePath"
+      if ($line -match $regexForMatchingShaAndPath) {
+        $line -replace "https:\/\/[^`"]*$regexForMatchingShaAndPath", "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+
+        $isUpdatedLines = $true
+        $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+        if (!$sdksInfo.ContainsKey($sdkpath)) {
+          $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
+          $sdksInfo.Add($sdkpath, $specReadmePath)
+        }
+      }
+      else {
+        $line
+      }
+    }
+    if ($isUpdatedLines) {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+function Test-PreviousScript() {
+  $result = $null
+  if ($LASTEXITCODE -eq 0) {
+    $result = "succeeded"
+  }
+  else {
+    $result = "failed"
+  }
+  Write-Host "Result: $result"
+  return $result
+}
+
+$packages = @()
+foreach ($sdkPath in $sdksInfo.Keys) {
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $result = Test-PreviousScript
+  if ($result -eq "succeeded") {
```

Comment:
You really don't even need this. PowerShell uses lazy boolean logic like JavaScript and C/C++. You can run your script and just check $LASTEXITCODE like so:

```
dotnet msbuild ...
if (!$LASTEXITCODE) {
  # continue with successful result...
}
```

There's also the automatic `$?` variable, but that can only be used *immediately* after the command you want to check (PowerShell or native/shell). Even as written, checking `$LASTEXITCODE` too long after you ran a native/shell command rists it being overwritten - it has all the same problems of Win32's `GetLastError()`, i.e. don't want to read it.